client/src/components/RegistrationForm.tsx

```diff
@@ -80,12 +104,32 @@ export function RegistrationForm({ value, handlePassword }: Props) {
           onChange={handlePassword}
           required
         />
-        <button
-          type="submit"
-          className="my-4 bg-black rounded w-full hover:bg-primary hover:text-black active:bg-secondary active:text-tertiary">
-          Create
-        </button>
+        {value.match(passwordValidRegex) && (
```

Comment:
This is OK, but it can confuse users why there is no Create button. They might search around for one before they realize it's because their password isn't valid. It's usually best to have the button there but disabled.